src/Compilers/CSharp/Portable/Symbols/Metadata/PE/TupleTypeDecoder.cs

```diff
@@ -159,6 +157,8 @@ private TypeSymbol DecodeType(TypeSymbol type)
             switch (type.Kind)
             {
                 case SymbolKind.ErrorType:
+                    _foundErrorType = true;
```

Comment:
>_foundErrorType = true; [](start = 20, length = 23)

I think that for the recovery purposes we should record presence of error types that have a use-site error. That is what we were trying to check with the ```HasUseSiteError``` above - the fact that we have a use-site error.